### DISCUSSION TITLE
Unexpected response from the LLM Server

### DISCUSSION THREAD

**Discussion by wilikeit** [pyscripter/pyscripter#1328] [Q&A]
Call prompt in pysripter:
![image](https://github.com/user-attachments/assets/5e4f6032-fa46-4640-a24e-b2dc96396b88)
Here is the configuration
![image](https://github.com/user-attachments/assets/0d32d520-85b8-4aac-abca-d986bf8e3e52)

<hr>

The same code model is normal in the chat window：
![image](https://github.com/user-attachments/assets/84f22573-606b-4178-93c4-c290e0b0a51b)

If I call it with curl, it will return as normal：
[root@localhost ~]# curl http://localhost:11434/api/generate -d '{
>   "model": "codellama:latest",
>   "prompt": "Fix bugs for the following code\na=1\ndef demo():\n  a=2",
>   "stream": false
> }'
{"model":"codellama:latest","created_at":"2024-10-02T02:31:47.860242353Z","response":"  The bug in this code is that it overwrites the value of `a` with 2. This means that when the function `demo()` is called, the value of `a` will be 2, not 1 as expected. To fix this bug, we can use a different variable name for the local variable inside the function to avoid overwriting the global variable `a`.\n\nHere's the corrected code:\n```\na = 1\ndef demo():\n    b = 2\n    print(b)\n```\nIn this version of the code, we have renamed the local variable `a` to `b`, so that it doesn't overwrite the global variable `a`.","done":true,"done_reason":"stop","context":[518,25580,29962,3532,14816,29903,29958,5299,829,14816,29903,6778,13,13,29943,861,24557,363,278,1494,775,13,29874,29922,29896,13,1753,13455,7295,13,29871,263,29922,29906,518,29914,25580,29962,13,29871,450,6494,297,445,775,338,393,372,975,8231,267,278,995,310,421,29874,29952,411,29871,29906,29889,910,2794,393,746,278,740,421,17482,2555,338,2000,29892,278,995,310,421,29874,29952,674,367,29871,29906,29892,451,29871,29896,408,3806,29889,1763,2329,445,6494,29892,591,508,671,263,1422,2286,1024,363,278,1887,2286,2768,278,740,304,4772,975,16554,278,5534,2286,421,29874,1412,13,13,10605,29915,29879,278,24114,775,29901,13,28956,13,29874,353,29871,29896,13,1753,13455,7295,13,1678,289,353,29871,29906,13,1678,1596,29898,29890,29897,13,28956,13,797,445,1873,310,278,775,29892,591,505,19533,278,1887,2286,421,29874,29952,304,421,29890,1673,577,393,372,1838,29915,29873,26556,278,5534,2286,421,29874,1412],"total_duration":67424212566,"load_duration":16864745,"prompt_eval_count":33,"prompt_eval_duration":6585351000,"eval_count":149,"eval_duration":60780171000}[root@localhost ~]#

**Comment by pyscripter** [comment]
Could you please clarify the steps required to reproduce this?

**Reply by wilikeit** [reply]
I think I found the problem，ollama is installed on a VM， I use nginx forwarded to "http://localhost:11434/api/generate", but  nginx configuration timeout is too short, lead to request didn't complete, ollama 500. 
<b>It worked fine after I increased the timeout parameter.</b>

INFO [main] model loaded | tid="140113544886144" timestamp=1727838855
time=2024-10-02T11:14:15.104+08:00 level=INFO source=server.go:572 msg="llama runner started in 1.55 seconds"
[GIN] 2024/10/02-11:15:13 | 500 | 1m0s | 192.168.1.102 | POST "/api/generate"
<span style="color:red"> </span>

◆  succesful
It was a surprise success  :)  
```
server {
    listen 8185;
        server_name _;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;

    location / {
proxy_pass http://127.0.0.1:11434;
        proxy_read_timeout 600s;  # Increase the timeout period to resolve the problem
        proxy_send_timeout 600s;  
    }
}
```
![image](https://github.com/user-attachments/assets/5f0cf0fe-855c-4d4f-a384-c738b803a66a)

**Comment by wilikeit** [comment]
I would like to ask you a question, please. In pyscripter，What parameters are sent to codellama by clicking on Fixed Bugs? 

Here's what I did. 
curl http://localhost:11434/api/generate -d '{

"model": "codellama:latest", 
"prompt": "Fix bugs for the following code\na=1\ndef demo():\n a=2",   #Is it reasonable?
"stream": false
} '

![image](https://github.com/user-attachments/assets/46875e54-6178-473a-93b2-ffb8b6d7ea26)

**Reply by pyscripter** [reply]
See https://github.com/pyscripter/pyscripter/blob/master/Source/uLLMSupport.pas#L812

**Reply by wilikeit** [reply]
Thank you. I'll read it.
